muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application/modify-area-dialog.tsx

```diff
@@ -98,31 +98,37 @@ class DiscussionModifyArea extends SessionStateComponent<DiscussionModifyAreaPro
     
     let content = (closeDialog: ()=>any) => [
       (
-      <div className="container container--new-discussion-area-title">          
-        <input key="1" type="text" className="form-field form-field--new-discussion-area-name"
-        placeholder={this.props.i18n.text.get('plugin.discussion.createarea.name')}
-        value={this.state.name} onChange={this.onNameChange} autoFocus/>
+      <div className="container container--new-discussion-area-title">    
+      <div className="environment-dialog__form-element--wrapper">  
+        <div className="environment-dialog__form-element-label">{this.props.i18n.text.get('plugin.discussion.createarea.name')}</div>          
+          <input key="1" type="text" className="environment-dialog__form-element environment-dialog__form-element--new-discussion-area-name"
+          placeholder={this.props.i18n.text.get('plugin.discussion.createarea.name')}
+          value={this.state.name} onChange={this.onNameChange} autoFocus/>
+        </div>
       </div>    
       
       ),
 
       (
        <div className="container container--new-discussion-area-description">          
-         <textarea key="2" className="form-field form-field--new-discussion-area-description"
+         <div className="environment-dialog__form-element--wrapper">  
+           <div className="environment-dialog__form-element-label">{this.props.i18n.text.get('plugin.discussion.createarea.description')}</div>          
+           <textarea key="2" className="environment-dialog__form-element environment-dialog__form-element--new-discussion-area-description"
         onChange={this.onDescriptionChange} value={this.state.description}/>
+         </div>
        </div>
     )]
        
     let footer = (closeDialog: ()=>any)=>{
       return (          
-         <div className="jumbo-dialog__button-container">
-          {this.recovered ? <Button buttonModifiers="danger" onClick={this.clearUp} disabled={this.state.locked}>
-            {this.props.i18n.text.get('clear draft')}
-          </Button> : null}
-          <Button buttonModifiers={["warn", "standard-cancel"]} onClick={closeDialog} disabled={this.state.locked}>
+         <div className="environment-dialog__button-container">
+          <Button className="button button-dialog--clear" onClick={this.clearUp} disabled={this.state.locked}>
+            {this.props.i18n.text.get('plugin.discussion.createarea.clearDraft')}
+          </Button>
+          <Button className="button button-dialog--cancel" onClick={closeDialog}>
             {this.props.i18n.text.get('plugin.discussion.createarea.cancel')}
           </Button>
-          <Button buttonModifiers="standard-ok" onClick={this.modifyArea.bind(this, closeDialog)} disabled={this.state.locked}>
+          <Button className="button button-dialog--execute" onClick={this.modifyArea.bind(this, closeDialog)} disabled={this.state.locked}>
```

Comment:
This seems to be wrong regarding BEM. Button and button-dialog are both BEM blocks and it seems they are used as mixed blocks, however button-dialog block has only modifier class name which is wrong.

If we use modifiers we need to include the base block class name too as modifier is an extra class name and it cannot be the only class name of the html element.